### PR TITLE
Added UK: Cable OTA EPG Grabber

### PR DESCRIPTION
**Pull request by dollis-developer**

### CODE REVIEW

**Comment by dollis-developer** [comment]
This is based on the openPli patch by LraiZer

[https://github.com/OpenViX/enigma2/commit/ad6340850ac3e67e5c457d934fed9ae249faede0](url)
[https://forums.openpli.org/topic/33839-virgen-epg-patch-for-openpli/](url)

File: src/epggrab/otamux.c [hunk 1]
```diff
@@ -492,6 +492,7 @@ epggrab_ota_kick_cb ( void *p )
     [MM_EPG_ONLY_PSIP]               = "psip",
     [MM_EPG_ONLY_UK_FREESAT]         = "uk_freesat",
     [MM_EPG_ONLY_UK_FREEVIEW]        = "uk_freeview",
+    [MM_EPG_ONLY_UK_CABLE]           = "uk_cable",
```
Comment (perexg) [inline src/epggrab/otamux.c]: It seems like Virgin only specific thing, so I would probably prefer uk_cable_virgin name.

File: src/epggrab/module/eit.c [hunk 2]
```diff
@@ -854,7 +854,7 @@ _eit_callback
   /* Get transport stream */
   // Note: tableid=0x4f,0x60-0x6f is other TS
   //       so must find the tdmi
-  if (tableid == 0x4f || tableid >= 0x60) {
+  if (tableid == 0x4f || tableid >= 0x60 || tableid == 0x50 || tableid == 0x4E) {
```
[the remark this answers is not in the document]
Comment (dollis-developer) [reply src/epggrab/module/eit.c]: I must admit, I don't completely understand the code, however, before I added the above there were a lot of the following errors in the tbl-eit trace log and only 24 hours EPG was loading:

2017-10-01 18:30:10.900 [  TRACE]:tbl-eit: uk_cable: invalid tsid found tid 0x50, onid:tsid 61472:25 != 61472:12
2017-10-01 18:30:10.903 [  TRACE]:tbl-eit: uk_cable: invalid tsid found tid 0x4E, onid:tsid 61472:25 != 61472:104

I traced that back to this section of the code from the error message and when I added the change the errors went away and full EPG was pulled down from the one mux that carries it.

[the remark this answers is not in the document]
Comment (perexg) [reply src/epggrab/module/eit.c]: The 'actual' data should be for the 'actual' transponder (TSID + ONID). It seems that the broadcaster does not care about the DVB specification and transmits the EPG data for services in other transponders in the 'actual' transpoder table IDs.

In this case, I would suggest to use 'spec' variable to handle this specific requirement in the epg parser. So, define something like 'EIT_SPEC_UK_CABLE_VIRGIN' and add a special case to the code.

**Comment by perexg** [comment]
Merged. Thanks.

**Pull request closed without merging**